### PR TITLE
Add PostgreSQL, MongoDB Atlas & MongoDB Replica Set Support

### PR DESCRIPTION
**Pull request by ProdPreva1l**
Adds support for PostgreSQL.
Uses all better native postgre alternatives
Adds support for srv record based mongodb atlas

### CODE REVIEW

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
Excellent work - could you update docs?

**Comment by ProdPreva1l** [comment]
Passing all tests, tested for Atlas and Postgres, also adds a safety check for UUID types on mongodb

**Comment by ProdPreva1l** [comment]
dont merge, resolving an issue, appears to be that sasl1 auth is goofd

File: common/src/main/java/net/william278/husksync/database/MongoDbDatabase.java [hunk 1]
```diff
@@ -64,14 +65,22 @@ public MongoDbDatabase(@NotNull HuskSync plugin) {
...
-                    credentials.getUsername(),
-                    credentials.getPassword(),
-                    credentials.getDatabase(),
-                    credentials.getMongoAuthDb()
-            );
+            if (plugin.getSettings().getDatabase().getType() == Type.MONGO_ATLAS) {
+                String baseURI = "mongodb+srv://{0}:{1}@{2}/?retryWrites=true&w=majority";
+                baseURI = baseURI.replace("{0}", credentials.getUsername());
+                baseURI = baseURI.replace("{1}", credentials.getPassword());
+                baseURI = baseURI.replace("{2}", credentials.getHost());
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/husksync/database/MongoDbDatabase.java:72]:
```suggestion
            final Database.Type type = plugin.getSettings().getDatabase().getType();
            if (type == Type.MONGO_ATLAS) {
                final String uri = String.format("%s://%s:%s@%s/?retryWrites=true&w=majority",
                type.getProtocol(), credentials.getUsername(), credentials.getPassword(), credentials.getHost());

File: common/src/main/java/net/william278/husksync/database/Database.java [hunk 2]
```diff
@@ -258,7 +257,9 @@ public final void pinSnapshot(@NotNull User user, @NotNull UUID versionUuid) {
     public enum Type {
         MYSQL("MySQL", "mysql"),
         MARIADB("MariaDB", "mariadb"),
-        MONGO("MongoDB", "mongo");
+        POSTGRES("PostgreSQL", "postgresql"),
+        MONGO("MongoDB", "mongo"),
+        MONGO_ATLAS("MongoDB", "atlas");
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/husksync/database/Database.java:262]:
```suggestion
        MONGO_ATLAS("MongoDB", "mongodb+srv");
```

File: docs/Setup.md [hunk 3]
```diff
@@ -24,8 +24,18 @@ This will walk you through installing HuskSync on your network of Spigot servers
 <summary><b>For MongoDB Users</b></summary>
```
Comment (WiIIiam278) [inline docs/Setup.md:24]: Could we mention postgres in Step 3? i.e.
```markdown
### 3. Enter Database & Redis server credentials
- Navigate to the HuskSync config file on each server (`~/plugins/HuskSync/config.yml`)
- Under `credentials` in the `database` section, enter the credentials of your database (MySQL/MariaDB/MongoDB/PostgreSQL). You shouldn't typically need to modify the `connection_pool` properties.
```

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]

File: common/src/main/java/net/william278/husksync/database/MongoDbDatabase.java (hunk 1 above)
Comment (ProdPreva1l) [reply common/src/main/java/net/william278/husksync/database/MongoDbDatabase.java]: I was going to switch to String.format, but my way makes more sense in the new implementation

File: common/src/main/java/net/william278/husksync/database/Database.java (hunk 2 above)
Comment (ProdPreva1l) [reply common/src/main/java/net/william278/husksync/database/Database.java]: This has been removed in favour of a mong_settings configuration section

File: docs/Setup.md (hunk 3 above)
Comment (ProdPreva1l) [reply docs/Setup.md:24]: done 👍

**Comment by ProdPreva1l** [comment]
**Breaks config for MongoDB users only**

**Comment by WiIIiam278** [comment]
Nice work. Thats a better way of doing mongo :)

**Pull request merged**